services/app-api/postgres/findAllSubmissionsWithRevisions.ts

```diff
@@ -0,0 +1,47 @@
+import { PrismaClient } from '@prisma/client'
+import { Submission2Type } from '../../app-web/src/common-code/domain-models'
+import {findAllSubmissionWrapper} from './findAllSubmissions'
+import {
+    isStoreError,
+    StoreError,
+} from './storeError'
+import { convertToSubmission2Type } from './submissionWithRevisionsHelpers'
+import {
+    getCurrentRevision,
+} from './submissionWithRevisionsHelpers'
+
+
+
+export async function findAllSubmissionsWithRevisions(
+    client: PrismaClient,
+    stateCode: string
+): Promise<Submission2Type[] | StoreError> {
+    const findResult = await findAllSubmissionWrapper(client, stateCode)
+
+    if (isStoreError(findResult)) {
+        return findResult
+    }
+
+    if (findResult === undefined) {
+        return findResult
+    }
+
+    const submissions: Submission2Type[] = []
+    const errors: Error | StoreError[] = []
+    findResult.forEach((submissionWithRevisions) => {
+        // check for current revision, if it doesn't exist, log an error
+        const currentRevisionOrError = getCurrentRevision(
+            submissionWithRevisions.id,
+            submissionWithRevisions
+        )
+        if (isStoreError(currentRevisionOrError)) {
+            // TODO: what should I do with these
```

Comment:
^ question for @macrael

Reply:
It doesn’t look like you are actually using the currentRevisionOrError if it’s not an error, so this may be moot, but it’s a good question, especially for queries that return multiple objects. When getCurrentRevision errors it’s a real programming error, submissions should always have at least one revision associated with them. I think it would be reasonable to log it and return an error in this case, ultimately throwing an ApolloError in the handler.